localstack-core/localstack/services/cloudwatch/provider_v2.py

```diff
@@ -835,6 +831,84 @@ def _get_timestamp(input: dict):
             history = [h for h in history if (date := _get_timestamp(h)) and date <= end_date]
         return DescribeAlarmHistoryOutput(AlarmHistoryItems=history)
 
+    def _evaluate_composite_alarms(self, context: RequestContext, triggering_alarm):
+        store = self.get_store(context.account_id, context.region)
+        alarms = list(store.alarms.values())
+        composite_alarms = [a for a in alarms if isinstance(a, LocalStackCompositeAlarm)]
+        for composite_alarm in composite_alarms:
```

Comment:
I haven't added any parent-child relationship between alarms for the first iteration. Therefore, all composite alarms needs to be evaluated if any metric alarm changes its state, regardless of whether that metric alarm is a part of a composite one. It should be ok for the Localstack use case, usually there aren't many alarms running locally so evaluation is not a heavy process. However, it puts some limitations like not being able to include a composite alarm into another composite alarm. Should something similar be needed in the future it is likely a parent-child relationship should be implemented.